vpc-architecture.yaml

```diff
@@ -1,17 +1,49 @@
 AWSTemplateFormatVersion: "2010-09-09"
 Description: "AWS VPC with Public and Private Subnets, Internet Gateway, NAT Gateway, and Routing"
 
+Parameters:
+  VpcCidrBlock:
+    Description: CIDR block for the VPC
+    Type: String
+    Default: 172.16.0.0/16
+    AllowedPattern: ^(([0-9]{1,3}\.){3}[0-9]{1,3}\/([0-9]|[1-2][0-9]|3[0-2]))$
+    ConstraintDescription: Must be a valid CIDR block (e.g., 172.16.0.0/16).
+  
+  EnableDnsSupport:
+    Description: Enable DNS support for the VPC
+    Type: String
+    Default: "true"
+    AllowedValues: 
+      - "true"
+      - "false"
+    ConstraintDescription: Must be "true" or "false".
+  
+  EnableDnsHostnames:
```

Comment:
No need to paramterize this value, rather it's better to pramterize all subnets cidr